service/src/main/java/uk/gov/hmcts/reform/fpl/service/email/content/OrderIssuedEmailContentProvider.java

```diff
@@ -0,0 +1,79 @@
+import static uk.gov.service.notify.NotificationClient.prepareUpload;
+
+@Slf4j
+@Service
+public class OrderIssuedEmailContentProvider extends AbstractEmailContentProvider {
+
+    private final LocalAuthorityNameLookupConfiguration localAuthorityNameLookupConfiguration;
+    private final RepresentativeService representativeService;
+    private final ObjectMapper objectMapper;
+
+    public OrderIssuedEmailContentProvider(@Value("${ccd.ui.base.url}") String uiBaseUrl,
+                                           ObjectMapper objectMapper,
+                                           HearingBookingService hearingBookingService,
+                                           LocalAuthorityNameLookupConfiguration localAuthorityNameLookupConfiguration,
+                                           DateFormatterService dateFormatterService,
+                                           RepresentativeService representativeService) {
+        super(uiBaseUrl, dateFormatterService, hearingBookingService);
+        this.objectMapper = objectMapper;
+        this.representativeService = representativeService;
+        this.localAuthorityNameLookupConfiguration = localAuthorityNameLookupConfiguration;
+    }
+
+    public Map<String, Object> buildOrderNotificationParametersForHmctsAdmin(final CaseDetails caseDetails,
+                                                                             final String localAuthorityCode,
+                                                                             final byte[] documentContents) {
+        CaseData caseData = objectMapper.convertValue(caseDetails.getData(), CaseData.class);
+        List<Representative> representativesServedByPost = representativeService.getRepresentativesByServedPreference(
+            caseData.getRepresentatives(), POST);
+        List<String> formattedRepresentatives = formatRepresentativesForPostNotification(representativesServedByPost);
+
+        return ImmutableMap.<String, Object>builder()
+            .put("needsPosting", !representativesServedByPost.isEmpty() ? "Yes" : "No")
+            .put("doesNotNeedPosting", representativesServedByPost.isEmpty() ? "Yes" : "No")
+            .put("courtName", localAuthorityNameLookupConfiguration.getLocalAuthorityName(localAuthorityCode))
+            .putAll(caseUrlOrDocumentLink(!representativesServedByPost.isEmpty(), documentContents,
+                caseDetails.getId()))
+            .put("respondentLastName", getFirstRespondentLastName(caseData.getRespondents1()))
+            .put("representatives", formattedRepresentatives.isEmpty() ? "" : formattedRepresentatives)
+            .build();
+    }
+
+    private Map<String, Object> caseUrlOrDocumentLink(boolean needsServing,
+                                                      final byte[] documentContents,
+                                                      Long caseId) {
+        ImmutableMap.Builder<String, Object> url = ImmutableMap.builder();
```

Comment:
why is this a map? variable name seems wrong

Reply:
think name is fine - soon to be changed to just be caseURL when we have print&post